src/cmd/cloud.js

```diff
@@ -41,311 +43,381 @@ const PLATFORMS = extend(utilities.knownPlatforms(), {
+					});
+			})
+			.catch((error) => {
+				const message = `Failed to flash ${device}`;
+				throw createAPIErrorResult({ error, message });
 			});
-		}).catch((err) => {
-			throw new VError(ensureError(err), 'Flash device failed');
-		});
 	}
 
-	_doFlash({ api, deviceId, fileMapping, targetVersion }) {
-		return Promise.resolve().then(() => {
-			return api.flashDevice(deviceId, fileMapping, targetVersion);
-		}).then((resp) => {
-			if (resp.status || resp.message) {
-				console.log('Flash device OK: ', resp.status || resp.message);
-			} else if (resp.output === 'Compiler timed out or encountered an error') {
-				console.log('\n' + (resp.errors && resp.errors[0]));
-				throw new Error('Compiler encountered an error');
-			} else {
-				throw api.normalizedApiError(resp);
-			}
-		}).catch(err => {
-			throw api.normalizedApiError(err);
-		});
+	_doFlash({ product, deviceId, fileMapping, targetVersion }){
+		return Promise.resolve()
+			.then(() => {
+				if (!product){
+					return;
+				}
+				this.ui.stdout.write(`marking device ${deviceId} as a development device${os.EOL}`);
+				return createAPI().markAsDevelopmentDevice(deviceId, true, product);
+			})
+			.then(() => {
+				this.ui.stdout.write(`attempting to flash firmware to your device ${deviceId}${os.EOL}`);
+				return createAPI().flashDevice(deviceId, fileMapping, targetVersion, product);
+			})
+			.then((resp) => {
+				if (resp.status || resp.message){
+					this.ui.stdout.write(`Flash device OK: ${resp.status || resp.message}${os.EOL}`);
+				} else if (resp.output === 'Compiler timed out or encountered an error'){
+					this.ui.stdout.write(`${os.EOL}${(resp.errors && resp.errors[0])}${os.EOL}`);
+					throw new Error('Compiler encountered an error');
+				} else {
+					throw normalizedApiError(resp);
+				}
+			})
+			.then(() => {
+				if (!product){
+					return;
+				}
+				[
+					`device ${deviceId} is now marked as a developement device and will NOT receive automatic product firmware updates.`,
+					'to resume normal updates, please visit:',
+					`https://console.particle.io/${product}/devices/unmark-development/${deviceId}`
+				].forEach(line => this.ui.stdout.write(`${line}${os.EOL}`));
+			})
+			.catch(err => {
+				throw normalizedApiError(err);
+			});
 	}
 
-	_flashKnownApp({ api, deviceId, filePath }) {
-		if (!settings.knownApps[filePath]) {
+	_flashKnownApp({ product, deviceId, filePath }){
+		if (!settings.knownApps[filePath]){
 			throw new VError(`I couldn't find that file: ${filePath}`);
 		}
 
-		return api.getAttributes(deviceId).then((attrs) => {
-			const spec = _.find(specs, { productId: attrs.product_id });
-			if (spec) {
-				if (spec.knownApps[filePath]) {
-					return api._populateFileMapping( { list: [spec.knownApps[filePath]] } );
-				}
+		return createAPI().getDeviceAttributes(deviceId)
+			.then((attrs) => {
+				const spec = _.find(specs, { productId: attrs.product_id });
```

Comment:
I think this should be `attrs.platform_id` for product devices.

Reply:
✅ in https://github.com/particle-iot/particle-cli/pull/565/commits/934d2e14bf3ce2ee5cf1d295117107fc939be7e0